### PR TITLE
Cleaned up test suite for tests in demo and enc/models

### LINKED ISSUE #509: Test suite cleanup
When we migrated from `unittest` to `pytest`, our understanding was that we had to use only test functions rather than test cases with test methods. This lead to structures like this:

https://github.com/pystiche/pystiche/blob/65f4d787e44b1ffbf7e5b6e48298ed8c7460e5a9/tests/integration/enc/test_multi_layer_encoder.py#L50-L68

In general, there are quite a few places were the test suite looks like

```python
def test_Foo_bar():
    ...

def test_Foo_baz():
    ...
```

instead of 

```python
class TestFoo:
    def test_bar(self):
        ...

    def test_baz(self):
        ...
```

Cleaning this up is a little bit of busy work, but can be a good entry point to the development of `pystiche`.

### PR DESCRIPTION
**Pull request by DevPranjal**
Partially fixes #509 
Few files added to keep the PR small and reviewable.

### CODE REVIEW

File: tests/integration/demo/test_demo.py [hunk 1]
```diff
@@ -5,44 +5,45 @@
...
-def test_transformer():
-    torch.manual_seed(0)
-    input_image = torch.rand(1, 3, 256, 256)
-    transformer = demo.transformer()
-
-    output_image = transformer(input_image)
-
-    assert output_image.shape == input_image.shape
+class TestDemoImages:
+    def test_images_smoke(self):
```
Comment (pmeier) [inline tests/integration/demo/test_demo.py:9]:
```suggestion
    def test_smoke(self):
```

File: tests/integration/demo/test_demo.py [hunk 2]
```diff
@@ -5,44 +5,45 @@
...
-
-def test_transformer():
-    torch.manual_seed(0)
-    input_image = torch.rand(1, 3, 256, 256)
-    transformer = demo.transformer()
-
-    output_image = transformer(input_image)
-
-    assert output_image.shape == input_image.shape
+class TestDemoImages:
```
Comment (pmeier) [inline tests/integration/demo/test_demo.py:8]:
```suggestion
class TestImages:
```

File: tests/integration/demo/test_demo.py [hunk 3]
```diff
@@ -5,44 +5,45 @@
...
+
+    @pytest.mark.parametrize(
+        ("name", "regions"),
+        [
+            ("castle", ("building", "water", "sky")),
+            ("church", ("building", "sky")),
+            ("cliff", ("water",)),
+        ],
+    )
+    def test_images_guides_smoke(self, name, regions):
```
Comment (pmeier) [inline tests/integration/demo/test_demo.py:32]:
```suggestion
    def test_guides_smoke(self, name, regions):
```

File: tests/integration/demo/test_demo.py [hunk 4]
```diff
@@ -5,44 +5,45 @@
...
+
+class TestDemoTransformers:
+    def test_transformer(self):
+        torch.manual_seed(0)
+        input_image = torch.rand(1, 3, 256, 256)
+        transformer = demo.transformer()
+
+        output_image = transformer(input_image)
+
+        assert output_image.shape == input_image.shape
```
Comment (pmeier) [inline tests/integration/demo/test_demo.py:49]: Could you revert this? We don't need a test class for a single test. If there will be more tests in the future, we can always add it later.

File: tests/integration/enc/models/test_vgg.py [hunk 5]
```diff
@@ -30,101 +30,104 @@ def vgg_multi_layer_encoder_loaders(vgg_archs):
...
-    for arch, loader in zip(vgg_archs, vgg_multi_layer_encoder_loaders):
-        with subtests.test(arch=arch):
-            asset = enc_asset_loader(arch)
-
-            multi_layer_encoder = loader(
-                pretrained=True,
-                weights="torch",
-                preprocessing=False,
-                allow_inplace=False,
+class TestVGGPretrained:
```
Comment (pmeier) [inline tests/integration/enc/models/test_vgg.py:33]: This class should be about the VGG nets. Right now we only have tests that include the pretrained variant, but that should not be exclusive.

```suggestion
class TestVGG:
```

File: tests/integration/enc/models/test_vgg.py [hunk 6]
```diff
@@ -30,101 +30,104 @@ def vgg_multi_layer_encoder_loaders(vgg_archs):
...
+    @pytest.mark.slow
+    def test_main(self, mocker):
```
Comment (pmeier) [inline tests/integration/enc/models/test_vgg.py:35]:
```suggestion
    def test_pretrained(self, mocker):
```

File: tests/integration/enc/models/test_vgg.py [hunk 7]
```diff
@@ -30,101 +30,104 @@ def vgg_multi_layer_encoder_loaders(vgg_archs):
...
+            mocks.make_mock_target(
+                "enc", "models", "vgg", "hub", "load_state_dict_from_url"
+            ),
+            return_value=state_dict,
+        )
+
+        model = vgg(pretrained=True)
+        ptu.assert_allclose(model.state_dict(), state_dict)
+
+    def test_num_classes_mismatch(self):
```
Comment (pmeier) [inline tests/integration/enc/models/test_vgg.py:47]:
```suggestion
    def test_pretrained_num_classes_mismatch(self):
```

**Review by pmeier — COMMENTED** [review]
Thanks @DevPranjal for the PR. I have a few minor naming suggestions inline. Apart from that it seems with this change we are no longer hitting this line:

https://github.com/pystiche/pystiche/blob/b7d354c4baf0404f519cce66593a6cee968a8d33/pystiche/enc/models/utils.py#L140

From quick look, there is nothing in here that should change this. I'll review more thorough later. Feel free to also do so if you want.

File: tests/integration/enc/models/test_utils.py [hunk 8]
```diff
@@ -113,109 +112,108 @@ def collect_modules(self, inplace: bool):
...
-    multi_layer_encoder_state_dict,
-    other_multi_layer_encoder_state_dict,
-    multi_layer_encoder_state_dict_key_map,
-):
-    inv_key_map = {v: k for k, v in multi_layer_encoder_state_dict_key_map.items()}
-    inv_state_dict = OrderedDict(
-        (inv_key_map[name], tensor)
-        for name, tensor in other_multi_layer_encoder_state_dict.items()
-    )
+    def tets_load_state_dict(
```
Comment (DevPranjal) [inline tests/integration/enc/models/test_utils.py:148]: The coverage issue was due to this misspelling

Comment (DevPranjal) [reply tests/integration/enc/models/test_utils.py]: The coverage looks fine now :)

**Comment by pmeier** [comment]
> The coverage looks fine now :)

Awesome, thanks a lot @DevPranjal!

---

For the future: you can apply suggestions directly from the GitHub UI. You can either click "Commit suggestion" under every comment. This will create a commit for every suggestion. Otherwise you can also add them to a batch and apply them all together. This way you don't need replicate my comments on your local machine somehow.

**Review by pmeier — APPROVED** [review]

**Comment by DevPranjal** [comment]
Noted, thanks!

**Pull request merged**